pytorch_lightning/trainer/evaluation_loop.py

```diff
@@ -347,6 +354,22 @@ def _evaluate(
 
         return eval_results
 
+    def reduce_eval_ddp(self, eval_results):
```

Comment:
this traversal of nested types looks like what @justusschock did in metrics. maybe one could re-use his `apply_to_collection` from `utilitiles.apply_func.py`

Reply:
good point @justusschock how would i adapt that here?

Reply:
Quick brain-compiled code:
```python
eval_results = apply_to_collection(
    eval_results, 
    dtype=torch.Tensor, 
    function=reduce_eval_ddp, 
    world_size=self.world_size
)
```
with `reduce_eval_ddp `defined as 
```python
def reduce_eval_ddp(eval_results, world_size):
    dist.all_reduce(v, op=dist.reduce_op.SUM)
    v = v / world_size
    return v
```